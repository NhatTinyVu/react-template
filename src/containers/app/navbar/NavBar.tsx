import React from 'react';
import { Link, RouteComponentProps } from 'react-router-dom';
import useRouter from 'use-react-router';
import { Menu } from 'antd';
import { get, find } from 'lodash';
import { routes } from '../constants';
import styles from './NavBar.less';

const initialKey = get(routes, '[0].url', '');

const NavBar: React.FC = () => {
  const {
    location: { pathname },
  }: RouteComponentProps = useRouter();

  const selectedKey = get(
    routes.find(({ url }) => pathname === url) ||
      find(routes, { children: [{ pathname }] }),
    'url',
    initialKey,
  );

  return (
    <Menu
      className={styles.navbar}
      theme="dark"
      mode="horizontal"
      defaultSelectedKeys={[initialKey]}
      selectedKeys={[selectedKey]}
    >
      {routes.map(({ title, url }) => (
        <Menu.Item key={url}>
          <Link to={url}>{title}</Link>
        </Menu.Item>
      ))}
    </Menu>
  );
};

export default NavBar;
