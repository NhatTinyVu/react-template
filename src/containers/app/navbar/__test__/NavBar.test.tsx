import React from 'react';
import { shallow } from 'enzyme';
import * as useRouter from 'use-react-router';
import toJSON from 'enzyme-to-json';
import NavBar from '..';

jest
  .spyOn(useRouter, 'default')
  .mockImplementation((): any => ({ location: { pathname: '/react-router' } }));

describe('NavBar', () => {
  test('renders correctly', (): void => {
    const wrapper = shallow(<NavBar />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
