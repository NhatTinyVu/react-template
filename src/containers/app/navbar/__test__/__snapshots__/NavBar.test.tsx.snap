// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<Menu
  className="navbar"
  defaultSelectedKeys={
    Array [
      "/react-hooks",
    ]
  }
  mode="horizontal"
  selectedKeys={
    Array [
      "/react-hooks",
    ]
  }
  theme="dark"
>
  <MenuItem
    key="/react-hooks"
  >
    <Link
      to="/react-hooks"
    >
      React hooks
    </Link>
  </MenuItem>
</Menu>
`;
